version/string.go

```diff
@@ -4,37 +4,68 @@
 package version
 
 import (
+	"encoding/json"
 	"fmt"
 	"runtime"
+	"strconv"
 	"strings"
 )
 
 var (
 	// String is displayed when CLI arg --version is used
 	String string
 
+	// String is displayed when CLI arg --json-version is used
+	JSONString string
+
 	// GitCommit is set in the build script at compile time
 	GitCommit string
 )
 
+type namedVersion struct {
```

Comment:
Given that the motivation behind this PR is to allow for programmatic access to the version details, I think it's worth exporting this type (and its fields) so other code can unmarshal the JSON into it.

If you do export it, I think it might need a different name so it's not `version.NamedVersion` when used elsewhere. Even just `version.Named`? Mandatory [style guide reference](https://google.github.io/styleguide/go/decisions#repetition) :rofl:

Alternatively, an explicit version struct would really help in such a scenario, but that undoes the nice pattern you have for text marshalling.

Consider the above as an exploration of trade-offs for you to consider, not explicit requests to change anything.